hypothesis-python/src/hypothesis/ghostwriter.py

```diff
@@ -0,0 +1,192 @@
+        if args.element_strategies:
+            raise NotImplementedError("Expected to pass everything as kwargs")
+
+        for k, v in zip(kwargs.keys, kwargs.mapped_strategy.element_strategies):
+            given_strategies[k] = v
+
+    indexer = list(_get_params(f)).index
+    return OrderedDict(sorted(given_strategies.items(), key=lambda kv: indexer(kv[0])))
+
+
+def _make_test(
+    *funcs: Callable, ghost: str, test_body: str, except_: Tuple[Type[Exception], ...]
+) -> str:
+    # the common elements of each ghostwriter
+    given_strategies = _get_strategies(*funcs)
+    given_args = ", ".join(
+        "{}={!r}".format(k, v)
+        .replace("sampled_from([False, True])", "booleans()")
+        .replace("just(None)", "none()")
+        .replace(".filter(_can_hash)", "")  # hide the special case for Decimal('snan')
+        for k, v in given_strategies.items()
+    )
+    for name in st.__all__:
+        given_args = given_args.replace("{}(".format(name), "st.{}(".format(name))
+    imports = {f.__module__ for f in funcs}
+    if "typing." in given_args:  # pragma: no cover
+        # After improving the strategy-unwrapping logic, I no longer have
+        # examples where from_type(typing.XYZ) appears in reprs...
+        # but nor am I willing to bet that this will never occur for user code
+        # when the worst-case is a useless import vs an import error.
+        imports.add("typing")
+    # We discard "builtins." below - this is probably not particularly useful
+    # for user code, but important for making a good impression in demos.
+    imports.discard("builtins")
+
+    if except_:
+        exceptions = []
+        for ex in except_:
+            if ex.__name__ in dir(builtins):
+                exceptions.append(ex.__name__)
+            else:
+                imports.add(ex.__module__)
+                exceptions.append(ex.__module__ + "." + ex.__name__)
+        test_body = SUPPRESS_BLOCK.format(
+            test_body=indent(test_body, prefix="    "),
+            exceptions="(" + ", ".join(exceptions) + ")"
+            if len(exceptions) > 1
+            else exceptions[0],
+        )
+
+    argnames = list(given_strategies)
+    body = TEMPLATE.format(
+        nothing_hint=""
+        if "=st.nothing()" not in given_args
+        else "# TODO: replace st.nothing() with an appropriate strategy\n\n",
+        given_args=given_args,
+        test_kind=ghost,
+        func_name="_".join(f.__name__ for f in funcs),
+        arg_names=", ".join(argnames),
+        test_body=indent(test_body, prefix="    "),
+    ).replace("builtins.", "")
+
+    result = (
+        IMPORT_SECTION.format(
+            imports=("import " + "\nimport ".join(sorted(imports)) if imports else "")
+        )
+        + "\nfrom hypothesis import {}given, strategies as st\n\n".format(
+            "assume, " if except_ else ""
+        )
+        + body
+    )
+    try:
+        import black
```

Comment:
I wonder if we should just depend on black unconditionally for the use of ghostwriter. Having its quality of output silently degrade when black isn't installed seems suboptimal.

Reply:
How about we make `black` a dependency for the CLI extra, but maintain the conditional call?  The output typically changes very little, and I do value minimal dependencies for the Python API at least.

Reply:
Yeah that sounds very reasonable.